### ISSUE TITLE
Unable to change docs title

### ISSUE THREAD

**Issue by endiliey** [facebook/Docusaurus#713]
### Is this a bug report?

Yes

### Have you read the [Contributing Guidelines on issues](https://github.com/facebook/Docusaurus/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

### Environment

Platform Agnostic
Latest Docusaurus (master branch)

### Steps to Reproduce / Reproducible Demo

1. Use latest master branch of [Docusaurus](https://github.com/facebook/Docusaurus) & install
2. Start server
```
cd website
yarn start
```
3. Go to http://localhost:3000/docs/en/next/installation.html
4. Change `docs/getting-started-installation.md` title to 
`Installation asdf`

<img width="191" alt="change" src="https://user-images.githubusercontent.com/17883920/40767067-fd2078f0-64e3-11e8-9351-1e16347d6c58.PNG">

5. Stop the server and start it again with `yarn start`
6. Go to http://localhost:3000/docs/en/next/installation.html

### Expected Behavior

<img width="725" alt="title_change" src="https://user-images.githubusercontent.com/17883920/40767150-3a3b04b2-64e4-11e8-8cd0-1035d6d3874f.PNG">

### Actual Behavior

<img width="609" alt="title_no_change" src="https://user-images.githubusercontent.com/17883920/40767163-455c54ea-64e4-11e8-9b3a-602692349d16.PNG">

### Investigation
This happen after PR #710 
Prior to this, `i18n/en.json` is always newly generated
After this, `i18n/en.json` will persist and this is why title is not changing

But PR #710 is correct because it fixes bug from #158 that allow user allow overriding of english strings

I think instead of
https://github.com/facebook/Docusaurus/blob/fa20c93d5c1b5a9369cf1c5909955fc58f0dcb8e/lib/write-translations.js#L27-L31

They should store it in something like `i18n/en-override.json`

**Comment by JoelMarcey** [comment]
Should we just always delete `i18n/en.json` and then regenerate when starting `server.js` or running `generate.js`?

**Comment by endiliey** [comment]
What is intended from PR #158 is that user can override Docusaurus provided strings that a project may not like (for example ReasonReact would like to say "Edit" instead of "Edit this Doc" in English). So they can modify `i18en.json` directly

Related code:
https://github.com/facebook/Docusaurus/blob/fa20c93d5c1b5a9369cf1c5909955fc58f0dcb8e/lib/write-translations.js#L27-L31

https://github.com/facebook/Docusaurus/blob/fa20c93d5c1b5a9369cf1c5909955fc58f0dcb8e/lib/write-translations.js#L158-L164

But this introduces this bug because if I made a change to my title, old title from `i18en.json` exist & overrides `i18en.json` new title

If we always delete `i18en.json` that is just the same as not allowing to override strings and effectively deleting this code:
https://github.com/facebook/Docusaurus/blob/fa20c93d5c1b5a9369cf1c5909955fc58f0dcb8e/lib/write-translations.js#L27-L31

That's why there has been no issue with that code before #710 since October 2017

**Comment by endiliey** [comment]
Hi @rickyvetter,
Just wanted to ask what is the intended / right behavior should be like when you implement #158 ?

**Comment by rickyvetter** [comment]
Thanks for the analysis @endiliey! Very thorough. You're correct in my intent. The goal was to allow overriding of all Docusaurus provided strings with project-specific defaults. I think those Object.assigns should probably be removed in favor of explicit checking.

**Comment by JoelMarcey** [comment]
From a chat with @rickyvetter....

Delete - https://github.com/facebook/Docusaurus/blob/fa20c93d5c1b5a9369cf1c5909955fc58f0dcb8e/lib/write-translations.js#L158-L164 

The custom lines above that should check if it exists in the file before writing.

**Comment by endiliey** [comment]
I just checked Docusaurus repo & our documentation website at https://docusaurus.io/docs/en/translation.html

![screenshot_20180605-092708](https://user-images.githubusercontent.com/17883920/40949922-f123ae6a-68a2-11e8-9636-cfb2405becfd.png)

Found out that i18en.json is always ignored through .gitignore. 
https://github.com/facebook/Docusaurus/blob/master/.gitignore
So actually the piece of code I mentioned has never been used at all. Should we just delete the ability to override the string once and for all ? The user didn't even know that they can even override the string unless they look at Docusaurus code directly.

I can assure you that this is the safest choice, it removes this bug, we have lesser code, and it won't break any user website.

**Comment by JoelMarcey** [comment]
Hi @endiliey While we don't check in the `en.json` file into Docusuarus, it is used when doing the uploads to Crowdin for translations. `write-translations.js` creates and uses that file for that purpose.

**Comment by endiliey** [comment]
@joelmarcey that's true, but isn't it always a newly generated i18en.json since there is no local i18en.json in GitHub? 

We always publish through continuous integration in which it uses the github repo as source of truth and there is no i18en.json when `write-translations` is executed, so it generates new i18en.json & upload it to crowdin. So the codes i mentioned is never used in production

**Comment by lili21** [comment]
same issue here.

**Comment by JoelMarcey** [comment]
I don't want to put the `showstopper` label on this, but this is a very high-pri bug as 1.2.0 brings it out into plain view for people.

This should be the next thing fixed before anything else, imho.

**Comment by JoelMarcey** [comment]
I will look at this some tomorrow. If anyone can look beforehand, that would be great. Even if, short-term, we have to break allowing custom strings to make sure people can modify the header metadata (e.g. title), I think we should consider that.

**Comment by endiliey** [comment]
I have few alternatives
1. Don't allow custom strings. This is like 1.1.5

Delete
https://github.com/facebook/Docusaurus/blob/fa20c93d5c1b5a9369cf1c5909955fc58f0dcb8e/lib/write-translations.js#L27-L31

https://github.com/facebook/Docusaurus/blob/fa20c93d5c1b5a9369cf1c5909955fc58f0dcb8e/lib/write-translations.js#L158-L164

2. Allow custom string but maybe user has to explicitly name it `custom-en.json`

https://github.com/facebook/Docusaurus/blob/fa20c93d5c1b5a9369cf1c5909955fc58f0dcb8e/lib/write-translations.js#L27-L3q

**Comment by JoelMarcey** [comment]
@endiliey 

I think I kinda like option #2 after giving it some thought today. What we can do is load the generated `i18n/en.json` first and and get that in the `currentTranslations` array. Then load `i18n/custom-en.json` and either override current values in the array if there would be a duplicate or append to the array.

What do you think?

**Comment by endiliey** [comment]
Sounds great @joelmarcey. 

Are you going to work on it ? Otherwise, I can submit a PR when I'm back roughly after this week.

**Comment by JoelMarcey** [comment]
I am going to give it a run - try to get a PR before the weekend.

**Comment by vaibhavsingh97** [comment]
@endiliey @JoelMarcey Can I try to fix this bug 😄

**Comment by JoelMarcey** [comment]
Hi @vaibhavsingh97 - I have a PR #775 out for this already. If you would like to have a look and provide any additional suggestions, that would be welcome 😄
